episodes/14-plot-raster-data.Rmd

```diff
@@ -0,0 +1,295 @@
+3. A plot title.
+
+::: solution
+
+```{r}
+DSM_TUD_df <- DSM_TUD_df %>%
+  mutate(fct_elevation_6 = cut(`tud-dsm-5m`, breaks = 6))
+
+levels(DSM_TUD_df$fct_elevation_6)
+
+my_col <- terrain.colors(6)
+
+ggplot() +
+  geom_raster(data = DSM_TUD_df, aes(x = x, y = y,
+                                       fill = fct_elevation_6)) +
+  scale_fill_manual(values = my_col, name = "Elevation") +
+  coord_quickmap() +
+  xlab("X") +
+  ylab("Y") +
+  labs(title = "Elevation Classes of the Digital Surface Model (DSM)")
+```
+
+:::
+
+:::
+
+## Layering Rasters
+
+We can layer a raster on top of a hillshade raster for the same area, and use a transparency factor to create a 3-dimensional shaded effect. A hillshade is a raster that maps the terrain using light and shadow to create a 3D-looking image that you would see from above when viewing the terrain. We will add a custom colour, making the plot grey.
+
+First we need to read in our DSM hillshade data and view the structure:
+```{r}
+DSM_hill_TUD <- rast("data/tud-dsm-5m-hill.tif")
+DSM_hill_TUD
+```
+
+Next we convert it to a data frame, so that we can plot it using `ggplot2`:
+```{r}
+DSM_hill_TUD_df <- as.data.frame(DSM_hill_TUD, xy = TRUE)
+str(DSM_hill_TUD_df)
+```
+
+Now we can plot the hillshade data:
+```{r}
+ggplot() +
+  geom_raster(data = DSM_hill_TUD_df,
+              aes(x = x, y = y, alpha = `tud-dsm-5m-hill`)) + 
+  scale_alpha(range =  c(0.15, 0.65), guide = "none") + 
+  coord_quickmap()
+```
+
+::: callout
+
+# Data tip
+
+Turn off, or hide, the legend on a plot by adding `guide = "none"` to a `scale_<something>()` function or by setting `theme(legend.position = "none")`.
+
+The `alpha` value determines how transparent the colours will be (0 being transparent, 1 being opaque).
+
+:::
+
+We can layer another raster on top of our hillshade by adding another call to the `geom_raster()` function. Let’s overlay `DSM_TUD` on top of the `DSM_hill_TUD`.
+
+```{r}
+ggplot() +
+  geom_raster(data = DSM_TUD_df , 
+              aes(x = x, y = y, 
+                  fill = `tud-dsm-5m`)) + 
+  geom_raster(data = DSM_hill_TUD_df, 
+              aes(x = x, y = y, 
+                  alpha = `tud-dsm-5m-hill`)) +  
+  scale_fill_viridis_c(option = "H") +  
+  scale_alpha(range = c(0.15, 0.65), guide = "none") +  
+  ggtitle("Elevation with hillshade") +
+  coord_quickmap()
+```
+
+::: challenge
+
+Use the `tud-dtm.tif` and `tud-dtm-hill.tif` files from the `data` directory to create a Digital Terrain Model (DTM) map of the TU Delft area.
+
+Make sure to:
+
+- include hillshade in the maps,
+- label axes,
+- include a title for each map,
+- experiment with various alpha values and color palettes to represent the data.
+
+::: solution
+
+```{r}
+# import DTM
+DTM_TUD <- rast("data/tud-dtm-5m.tif")
+DTM_TUD_df <- as.data.frame(DTM_TUD, xy = TRUE)
+
+# DTM Hillshade
+DTM_hill_TUD <- rast("data/tud-dtm-5m-hill.tif")
+DTM_hill_TUD_df <- as.data.frame(DTM_hill_TUD, xy = TRUE)
+
+ggplot() +
```

Comment:
the final plot looks quite grey... is it ok?

Reply:
I know it is. Grey is typically used for hillshade... not that useful in our case. I am considering dropping this part as it is not very relevant for us.

Reply:
I removed that part.